macos: add tab index labels

gpanders opened the pull request:
I mentioned this on Discord but didn't get a response if this is a desired feature. I went ahead and made an attempt to implement it and it wasn't too difficult.

I don't have much (read: any) experience with macOS UI development, so if there's a better way to do this please tell me.

---

Adds labels to the window tabs that indicate the tab index and the key to press to activate that tab.

Screenshot:

<img width="1712" alt="Screenshot 2023-09-27 at 9 05 43 PM" src="https://github.com/mitchellh/ghostty/assets/8965202/795c3ed3-e587-4e3e-b281-7370c6a2ddd5">

Indices only go up to 9 (which is what is supported via the default keyboard shortcuts):

<img width="1712" alt="Screenshot 2023-09-27 at 9 09 27 PM" src="https://github.com/mitchellh/ghostty/assets/8965202/a4c4335e-ac1d-4a1b-a8a1-0ff9ebae0ab4">

gpanders committed:
macos: add tab index labels

mitchellh reviewed (commented):
This looks really good! Especially for someone who has never done macOS dev before. I left a couple comments. I'll give this a go in a bit. I'm happy to take over for the comments just let me know.

gpanders committed:
macos: rename indexTabs() to relabelTabs()

This is a better name because the tab label may not necessarily be an
index (if a user binds some non-numeric key, for instance).

gpanders committed:
macos: create modifier flags as OptionSet instead of from array

Nit picky change, but the OptionSet class (which NSEvent.ModifierFlags
is an instance of) has an API for incrementally setting values without
needing to create an array.

gpanders committed:
macos: add keyEquivalent overload to also return modifier string

gpanders committed:
macos: display keybinding instead of index in tab label
